mrmustard/training/optimizer.py

```diff
@@ -87,6 +87,7 @@ def minimize(
                 Learn more about how to use callbacks to have finer control of the optimization
                 process in the :mod:`.callbacks` module.
         """
+        math._euclidean_opt = None  # temporary workaround
```

Comment:
how do we ensure this is only temporary?

Reply:
I say temporary with the intent of coming back to this to support the non-legacy Adam optimizer. For now it's there to have our pipeline working